Jenkinsfile.release

```diff
@@ -29,8 +29,11 @@ node ('macos1'){
     stage('Git & Dependencies') {
       slackSend color: 'good', message: BRANCH_NAME + ' build started. ' + env.BUILD_URL
 
+      git([url: 'https://github.com/status-im/status-react.git', branch: 'develop'])
```

Comment:
@jeluard Not 100% but I think using 'branch' here messes up release branch target: https://github.com/status-im/status-react/issues/3430

I believe solution is to have a separate target, or figure out way to not get BRANCH_NAME be shadowed